src/app/modules/item/components/item-permissions/item-permissions.component.ts

```diff
@@ -1,13 +1,62 @@
-import { Component, Input } from '@angular/core';
+import { Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
 import { ItemData } from '../../services/item-datasource.service';
 import { WatchedGroup } from 'src/app/core/services/group-watching.service';
+import {
+  ProgressSelectValue
+} from '../../../shared-components/components/collapsible-section/progress-select/progress-select.component';
+import { generateCanViewValues } from '../permissions-edit-dialog/permissions-edit-dialog-texts';
+import { Permissions } from '../../../../shared/helpers/group-permissions';
+import { GroupPermissionsService } from '../../../../shared/http-services/group-permissions.service';
+import { ActionFeedbackService } from '../../../../shared/services/action-feedback.service';
+import { PermissionsInfo } from '../../helpers/item-permissions';
 
 @Component({
   selector: 'alg-item-permissions',
   templateUrl: './item-permissions.component.html',
   styleUrls: [ './item-permissions.component.scss' ],
 })
-export class ItemPermissionsComponent {
+export class ItemPermissionsComponent implements OnChanges {
+  @Output() changed = new EventEmitter<void>();
+
   @Input() itemData?: ItemData;
   @Input() watchedGroup?: WatchedGroup;
+
+  canViewValues: ProgressSelectValue<string>[] = generateCanViewValues('Groups');
+  permissionsDialog: 'closed' | 'opened' = 'closed';
```

Comment:
If it has 2 states only, why not using a boolean value?

Reply:
Fixed